clientconn.go

```diff
@@ -671,21 +671,23 @@ func (cc *ClientConn) Connect() {
 }
 
 // waitForResolvedAddrs blocks until the resolver has provided addresses or the
-// context expires.  Returns nil unless the context expires first; otherwise
-// returns a status error based on the context.
-func (cc *ClientConn) waitForResolvedAddrs(ctx context.Context) error {
+// context expires. It returns two values:
+//   - a boolean indicating if there was a name resolution delay.
+//   - an error which is nil unless the context expires first; otherwise returns
+//     a status error based on the context.
+func (cc *ClientConn) waitForResolvedAddrs(ctx context.Context) (bool, error) {
```

Comment:
why do you have a bool. Just check at the caller if err happend or not and set the flag accordingly. This is unnecessary

Reply:
Updated